### PR TITLE
Instance wide uncommitted queue

### LINKED ISSUE #370: Instance-wide awareness of uncommitted queue for concurrency
instance-wide awareness of uncommitted queue for concurrency warnings/blocks across developers working in separate feature branches/namespaces.

### PR DESCRIPTION
**Pull request by isc-hwojnick**
Fixes #370

### CODE REVIEW

File: cls/SourceControl/Git/Utils.cls [hunk 1]
```diff
@@ -2418,6 +2418,63 @@ ClassMethod UncommittedWithAction() As %Library.DynamicObject
...
+        set namespace = $ORDER(allNamespaces(namespace))
+        quit:namespace=""
+        set $NAMESPACE = namespace
+        set statement = ##class(%SQL.Statement).%New()
+        set status = statement.%Prepare(.gitscQuery)
+        set result = statement.%Execute()
+        do result.%Next()
+        set gitEnabled = result.%GetData(1)
+        if (gitEnabled) set enabledNamespaces = enabledNamespaces _ $listbuild(namespace)
+
```
Comment (isc-tleavitt) [inline cls/SourceControl/Git/Utils.cls:2455]: Nit: remove extra whitespace (blank lines)

**Review by isc-tleavitt — CHANGES_REQUESTED** [review]
One really tiny nitpick, otherwise good.

**Review by isc-tleavitt — CHANGES_REQUESTED** [review]
One tiny nit, and a request: please test this as a user with just %Developer to make sure it all works properly without doing extra SQL security work.

**Comment by isc-tleavitt** [comment]
Can retarget this to main at this point.

**Comment by isc-pbarton** [comment]
I made some fixes to get this working with %Developer roles, and a couple other minor improvements:

- The alert will no longer show up if I have the item edited in the current namespace
- There's now a setting in Configure I can use to turn off the alerts entirely

File: cls/SourceControl/Git/Change.cls [hunk 2]
```diff
@@ -185,6 +185,48 @@ ClassMethod RefreshUncommitted(Display = 0, IncludeRevert = 0, Output gitFiles,
...
+    set qHandle("i") = 1
+    new $namespace
+    while $LISTNEXT(namespaces, tPtr, tValue) {
+        set namespace = $ZCONVERT(tValue, "U")
+        set $NAMESPACE = namespace
+        set statement = ##class(%SQL.Statement).%New()
+        $$$ThrowOnError(statement.%Prepare(qHandle("q"), 0))
+        set resultSet = statement.%Execute()
+        throw:resultSet.%SQLCODE<0 ##class(%Exception.SQL).CreateFromSQLCODE(resultSet.%SQLCODE,resultSet.%Message)
+        while resultSet.%Next() {
```
Comment (isc-tleavitt) [inline cls/SourceControl/Git/Change.cls:206]: Should check byref %Status from %Next(.sc) in/after loop

File: cls/SourceControl/Git/Utils.cls [hunk 3]
```diff
@@ -2417,6 +2422,47 @@ ClassMethod UncommittedWithAction() As %Library.DynamicObject
...
+    quit enabledNamespaces
+}
+
+/// Returns true if the given item has uncommitted changes on a different namespace in this instance.
+ClassMethod InstanceWideUncommittedCheck(InternalName As %String, Output User, Output Namespace) As %Boolean
+{
+    set isUncommitted = 0
+    set resultSet = ##class(SourceControl.Git.Change).InstanceUncommittedFunc()
+    throw:resultSet.%SQLCODE<0 ##class(%Exception.SQL).CreateFromSQLCODE(resultSet.%SQLCODE,resultSet.%Message)
+    while resultSet.%Next() {
```
Comment (isc-tleavitt) [inline cls/SourceControl/Git/Utils.cls:2454]: Should check byref %Status from %Next(.sc) inside/outside of loop

File: cls/SourceControl/Git/Utils.cls [hunk 4]
```diff
@@ -2417,6 +2422,47 @@ ClassMethod UncommittedWithAction() As %Library.DynamicObject
...
+    }
+
+    quit enabledNamespaces
+}
+
+/// Returns true if the given item has uncommitted changes on a different namespace in this instance.
+ClassMethod InstanceWideUncommittedCheck(InternalName As %String, Output User, Output Namespace) As %Boolean
+{
+    set isUncommitted = 0
+    set resultSet = ##class(SourceControl.Git.Change).InstanceUncommittedFunc()
```
Comment (isc-tleavitt) [inline cls/SourceControl/Git/Utils.cls:2452]: Concern: is this going to hit SQL privilege issues?

**Review by isc-tleavitt — CHANGES_REQUESTED** [review]
Two nitpicks and one SQL security question.

**Review by isc-tleavitt — CHANGES_REQUESTED** [review]
Just a bit of housekeeping in CHANGELOG.md and module.xml needed here, then good to go.

**Review by isc-tleavitt — APPROVED** [review]

**Pull request merged**